### ISSUE TITLE
io cannot read h5 dataset having 'u1' dtype

### ISSUE THREAD

**Issue by Rendok** [tensorflow/io#483]
Hello! Thanks for your troubles.

_tf version: 2.0.0.rc0
tf_io version: 0.9.0-dev1759_

I got a bug trying to read an h5 data set. Here the code to illustrate a problem:

```
import h5py
import tensorflow_io.hdf5 as io
data = np.array([1, 1, 1, 1])

with h5py.File('demo.h5', 'w') as f:
    f.create_dataset('uint', data=data, dtype='u1')
    f.create_dataset('int', data=data, dtype='i4')
    f.create_dataset('float', data=data)


dataset = io.list_hdf5_datasets('demo.h5')
print(dataset)
```

As you can see, uint is missing.

Cheers

**Comment by yongtang** [comment]
@Rendok PR #486 has been created to address the issue.

I would also suggest you to take a look at use `tfio.IOTensor.from_hdf5` as it will be easier to work with in eager mode. (We may also deprecate list_hdf5_datasets once TF 2.0.0 is release).

Here is a usage example to work with `tfio.IOTensor.from_hdf5` (for tensorflow 2.0.0):

```python
import tensorflow
import tensorflow_io as tfio

data = np.array([1, 1, 1, 1])
with h5py.File(filename, 'w') as f:
  f.create_dataset('uint', data=data, dtype='u1')
  f.create_dataset('int', data=data, dtype='i4')
  f.create_dataset('float', data=data)

hdf5 = tfio.IOTensor.from_hdf5(filename)

# to get any hdf5 dataset
uint_data = hdf5('/uint')

# convert to a normal tensor
uint_tensor = uint_data.to_tensor()

# access a slice of hdf5 dataset as a normal tensor
uint_slice = uint_data[0:2]

```

**Comment by Rendok** [comment]
@yongtang Thanks!

Your code snippet doesn't work on my machine.

> tensorflow.python.framework.errors_impl.InvalidArgumentError: unsupported data type: 216172782113784254 [Op:HDF5IndexableInit]

However, if I comment `# f.create_dataset('uint', data=data, dtype='u1')`, it works. So the problem is with the support of uints.

Otherwise, thanks for the hint. I also can convert `hdf5('/int').to_dataset()`, which works perfectly fine.

**Comment by yongtang** [comment]
@Rendok The `u1` issue has been resolved in PR #486. If you pull the latest nightly the issue should have been fixed. Let me know if the issue still exist.
